src/components/dropdown-menu/DropdownMenu.tsx

```diff
@@ -32,30 +37,39 @@ interface DropdownMenuClickProps<T> extends DropdownMenuBaseProps<T> {
 interface DropdownMenuContextProps<T> extends DropdownMenuBaseProps<T> {
   trigger: 'contextMenu';
   children: ReactNode;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  as?: ElementTags | React.ComponentType<any>;
 }
 
 export type DropdownMenuProps<T> =
   | DropdownMenuContextProps<T>
   | DropdownMenuClickProps<T>;
 
-export function DropdownMenu<T>(props: DropdownMenuProps<T>) {
+export function DropdownMenu<T = unknown, E extends ElementType = 'div'>(
+  props: DropdownMenuProps<T> & { wrapperProps?: ElementProps<E> },
+) {
   const { trigger, ...otherProps } = props;
 
   if (trigger === 'contextMenu') {
-    return <DropdownContextMenu {...otherProps} />;
+    return <DropdownContextMenu<T, E> {...props} />;
   }
-
   return (
     <DropdownClickMenu {...otherProps}>{props.children}</DropdownClickMenu>
   );
 }
 
-const HandleMenuContextDiv = styled.div`
-  display: contents;
-`;
-
-function DropdownContextMenu<T>(props: Omit<DropdownMenuProps<T>, 'trigger'>) {
-  const { children, onSelect, ...otherProps } = props;
+function DropdownContextMenu<T, E extends ElementType = 'div'>(
+  props: Omit<DropdownMenuContextProps<T>, 'trigger'> & {
+    wrapperProps?: ElementProps<E>;
+  },
```

Comment:
Can you flatten the prop structure by putting the props from the `as` element at the first level? I don't think we will ever run into name clashes.

```suggestion
  props: Omit<DropdownMenuContextProps<T>, 'trigger'> & ElementProps<E>
```

Reply:
i did it like that because we have props for the `Menu` and others for `Wrapper` unless we destruct all the menu pros at the top level 
https://github.com/zakodium-oss/react-science/blob/5286bc0db1d645a94733a41574c0a94ba9543049/src/components/dropdown-menu/DropdownMenu.tsx#L109

Reply:
We can destructuring the `Menu` props but we have to take care of any coming changes on the component has to take care about this (I mean for the menu props)